lib/active_merchant/billing/gateway.rb

```diff
@@ -212,6 +212,16 @@ def user_agent
         })
       end
 
+      def strip_invalid_xml_chars(xml)
+        begin
+          REXML::Document.new(xml)
+        rescue REXML::ParseException
+          xml = xml.gsub(/&(?!(?:[a-z]+|#[0-9]+|x[a-zA-Z0-9]+);)/, '&amp;')
```

Comment:
If I'm following this correctly - if you have valid XML, you'll return a `REXML::Document` - but if you have invalid XML that raises a `REXML::ParseException`, you'll get a string back? Can you make the return values consistent?

This should also be going through Nokogiri; we'd like to use that over REXML going forward.